Documentation/DesignGuidelines.md

```diff
@@ -8,26 +8,72 @@ This document assumes basic familiarity
 with the features of ReactiveCocoa. The [Framework Overview][] is a better
 resource for getting up to speed on the functionality provided by RAC.
 
-**[The Signal contract](#the-signal-contract)**
+1. **[The `Event` contract](#the-event-contract)**
+1. **[The `Signal` contract](#the-signal-contract)**
+1. **[The `SignalProducer` contract](#the-signalproducer-contract)**
+1. **[Best practices](#best-practices)**
+1. **[Implementing new operators](#implementing-new-operators)**
 
-**[The SignalProducer contract](#the-signalproducer-contract)**
+## The `Event` contract
 
-**[Best practices](#best-practices)**
+Events are fundamental to ReactiveCocoa. Signals and signal producers both send
+events, and may be collectively called “event streams.”
 
-**[Implementing new operators](#implementing-new-operators)**
+Event streams must conform to the following grammar:
 
+```
+Next* (Interrupted | Error | Completed)?
+```
 
-## The Signal contract
+This states that an event stream consists of:
 
+ 1. Any number of `Next` events
+ 1. Optionally followed by one terminating event, which is any of `Interrupted`, `Error`, or `Completed`
 
-## The SignalProducer contract
+After a terminating event, no other events will be received.
 
+#### `Next`s provide values or indicate the occurrence of events
+#### Errors behave like exceptions and propagate immediately
```

Comment:
e.g. operators that could delay `Next` events (e.g. `delay`) propagate `Error` events immediately?